nixos/modules/services/databases/mongodb.nix

```
@@ -176,7 +176,7 @@ in
         postStart = ''
             if test -e "${cfg.dbpath}/.first_startup"; then
               ${optionalString (cfg.initialScript != null) ''
-                ${mongodb}/bin/mongo -u root -p ${cfg.initialRootPassword} admin "${cfg.initialScript}"
+                ${mongodb}/bin/mongo ${optionalString (cfg.initialRootPassword != null) "-u root -p ${cfg.initialRootPassword}"} admin "${cfg.initialScript}"
```

```suggestion
                ${mongodb}/bin/mongo ${optionalString (cfg.enableAuth) "-u root -p ${cfg.initialRootPassword}"} admin "${cfg.initialScript}"
```

I think that should use `enableAuth`, since `enableAuth` guarantees that a password is set, but not the other way around (the root user may not exist even though `initialRootPassword` is non-null.